### PR TITLE
feat: add new studyload inputs to ai competencies

### LINKED ISSUE #850: manually added competencies have wrong duration
With the new time format, we broke the time entry / parsing for manually added competencies:

Entry of 1:30 h:
Image

When added:
Image

After parsing and saving:
Image

### PR DESCRIPTION
**Pull request by Thiemann96**
closes #850 and #827  
 - when adding ai creation competencies have new input row for studyload
 - fixed a bug where the studyload was calculated by concatenating to strings (3hour 30 minutes would be 30030 in the studyload)

### CODE REVIEW

**Review by jona159 — COMMENTED** [review]
I didnt review the code yet, but in the preview i think there is no form validation on the studyload input fields of ai competencies. For example i can enter 70 minutes

File: src/app/issuer/components/badgeclass-edit-form/badgeclass-edit-form.component.html [hunk 1]
```diff
@@ -1,200 +1,215 @@
...
-						required: chooseABadgeCategory
-					}"
-					[autofocus]="true"
-					noTopMargin="true"
-				></oeb-select>
-			</div>
-			<div class="tw-italic tw-gap-2 md:tw-pl-8">
-				<div class="tw-flex tw-py-2">
-					<p hlmP size="sm" class="tw-text-purple">{{ 'Badge.competency' | translate }}-Badges&nbsp; </p>
-					<p hlmP size="sm" class="md:tw-hidden"> {{ 'CreateBadge.competencyBadgeShortInfo' | translate }}.</p>
```
Comment (timber-they) [inline src/app/issuer/components/badgeclass-edit-form/badgeclass-edit-form.component.html:27]: There seems to be quite a lot of unnecessary reformatting. Maybe avoid that for the future, since it makes spotting actual changes harder and imo doesn't provide any advantage, the way it was done here.

File: src/app/issuer/components/badgeclass-edit-form/badgeclass-edit-form.component.ts [hunk 2]
```diff
@@ -226,6 +227,8 @@ export class BadgeClassEditFormComponent extends BaseAuthenticatedRoutableCompon
 				// but since it doesn't make sense to remove the
```
Comment (timber-they) [inline src/app/issuer/components/badgeclass-edit-form/badgeclass-edit-form.component.ts:227]: The comments belong to the control below the ones you created. Also, is the `studyLoad` control actually still used, despite there being the new hours and minutes controls? Also, shouldn't at least one of the two new fields also be required to be a positive integer?

File: src/app/issuer/components/badgeclass-edit-form/badgeclass-edit-form.component.ts [hunk 3]
```diff
@@ -422,9 +425,7 @@ export class BadgeClassEditFormComponent extends BaseAuthenticatedRoutableCompon
 			// based on ai suggestions, they can't be separated anymore and thus will
 			// be displayed as competencies entered by hand
 			aiCompetencies: [],
-			competencies: badgeClass.extension['extensions:CompetencyExtension']
-				? competencies
-				: [],
+			competencies: badgeClass.extension['extensions:CompetencyExtension'] ? competencies : [],
```
Comment (timber-they) [inline src/app/issuer/components/badgeclass-edit-form/badgeclass-edit-form.component.ts:428]: I actually like shorter lines of code.

File: src/app/issuer/components/badgeclass-edit-form/badgeclass-edit-form.component.html (hunk 1 above)
Comment (Thiemann96) [reply src/app/issuer/components/badgeclass-edit-form/badgeclass-edit-form.component.html:27]: We'll be applying prettier to all files on the main branch on tuesday next week, so this will not be a problem from there on

File: src/app/issuer/components/badgeclass-edit-form/badgeclass-edit-form.component.ts (hunk 3 above)
Comment (Thiemann96) [reply src/app/issuer/components/badgeclass-edit-form/badgeclass-edit-form.component.ts:428]: Till tuesday next week you'll have a chance of editing the prettier config in `.prettierrc`to apply your preffered formatting. I do not really mind, as long as we keep it uniform

File: src/app/issuer/components/badgeclass-edit-form/badgeclass-edit-form.component.ts (hunk 2 above)
Comment (Thiemann96) [reply src/app/issuer/components/badgeclass-edit-form/badgeclass-edit-form.component.ts:227]: 1. ´studyload` control is not needed anymore. i deleted it alongside the comment 
2. i cant quite grasp your second question,  i added the `positiveIntegerOrNull()` validation, no?

**Comment by Thiemann96** [comment]
> I didnt review the code yet, but in the preview i think there is no form validation on the studyload input fields of ai competencies. For example i can enter 70 minutes

You're right. I forgot the validation in the ai tool. Added it now in 98de4860d1f7ce05b8a0088d523cc492293d9d69

File: src/app/issuer/components/badgeclass-edit-form/badgeclass-edit-form.component.ts (hunk 2 above)
Comment (timber-they) [reply src/app/issuer/components/badgeclass-edit-form/badgeclass-edit-form.component.ts]: @Thiemann96 yes, but that also allows 0. So currently it's possible to enter 0 for both fields, whereas before a duration of 0 was not allowed. I don't think it's entirely trivial though, because at least one field would have to be non-0, for which we don't have a validator yet.

File: src/app/issuer/components/badgeclass-edit-form/badgeclass-edit-form.component.html [hunk 4]
```diff
@@ -359,7 +355,11 @@ <h2 class="tw-whitespace-pre-wrap tw-text-sm tw-text-oebblack tw-font-bold">
 					</div>
 				</div>
 				<hr class="tw-my-8 tw-border tw-border-[var(--color-purple)] tw-border-dashed" />
-				<oeb-collapsible [trigger]="collapseCompetenciesTrigger" [defaultOpen]="collapsedCompetenciesOpen">
+				<oeb-collapsible
+					[id]="'competencies-by-hand-section'"
```
Comment (timber-they) [inline src/app/issuer/components/badgeclass-edit-form/badgeclass-edit-form.component.html:359]: Why the double quotes? Also, what does this change have to do with this ticket?

File: src/app/issuer/components/badgeclass-edit-form/badgeclass-edit-form.component.html [hunk 5]
```diff
@@ -431,6 +429,7 @@ <h2 class="tw-whitespace-pre-wrap tw-text-oebblack tw-font-bold">
 									class="tw-flex tw-flex-col md:tw-flex-row md:tw-justify-between tw-gap-5 md:tw-gap-0"
 								>
 									<oeb-input
+										id="{{ 'competencyTitle_' + i }}"
```
Comment (timber-they) [inline src/app/issuer/components/badgeclass-edit-form/badgeclass-edit-form.component.html:432]: I feel like multiple tickets are being mixed up here. Or what is the purpose of this change?

Comment (Thiemann96) [reply src/app/issuer/components/badgeclass-edit-form/badgeclass-edit-form.component.html:432]: these issues were caused because the main branch was already ahead again by Youssef's change with this commit baca52e  (because you're right this has nothing to do with this ticket 👍 )

I updated the branch since

File: src/app/issuer/components/badgeclass-edit-form/badgeclass-edit-form.component.ts (hunk 2 above)
Comment (Thiemann96) [reply src/app/issuer/components/badgeclass-edit-form/badgeclass-edit-form.component.ts]: Ah yeah of course, good catch. It is indeed not as trivial as I hoped, especially with dynamically added input fields for competencies.

**Comment by timber-they** [comment]
It seems this doesn't build. Also, please fix the merge conflicts.

**Review by jona159 — APPROVED** [review]
